src/natcap/invest/utils.py

```diff
@@ -43,6 +44,173 @@
+}
+GTE_0 = {
+    "validation_options": {
+        "expression": "value >= 0"
+    }
+}
+BETWEEN_0_AND_1 = {
+    "validation_options": {
+        "expression": "(value >= 0) & (value <= 1)"
+    }
+}
+
+DISTANCE = {
+    "type": "number",
+    "units": u.meter,
+    "validation_options": {
+        "expression": "value >= 0"
+    }
+}
+
+METER_RASTER = {
+    "type": "raster",
+    "bands": {
+        1: {
+            "type": "number",
+            "units": u.meter
+        }
+    }
+}
+
+AOI_ARG = {
+    **AREA,
+    "name": "area of interest",
+    "about": (
+        "A GDAL-supported vector file.  This AOI instructs "
+        "the model where to clip the input data and the extent "
+        "of analysis.  Users will create a polygon feature "
+        "layer that defines their area of interest.  The AOI "
+        "must intersect the Digital Elevation Model (DEM)."),
+}
+LULC_ARG = {
+    "type": "raster",
+    "bands": {1: {"type": "code"}},
+    "about": (
+        "A GDAL-supported raster file containing LULC code "
+        "(expressed as integers) for each cell."),
+    "name": "land use/land cover"
+}
+DEM_ARG = {
+    "type": "raster",
+    "bands": {
+        1: {
+            "type": "number",
+            "units": u.meter
+        }
+    },
+    "about": (
+        "A GDAL-supported raster file containing elevation values for "
+        "each cell.  Make sure the DEM is corrected by filling in "
+        "sinks, and if necessary burning hydrographic features into "
+        "the elevation model (recommended when unusual streams are "
+        "observed.) See the Working with the DEM section of the "
+        "InVEST User's Guide for more information."),
+    "name": "digital elevation model"
+}
+PRECIP_ARG = {
+    "type": "raster",
+    "bands": {
+        1: {
+            "type": "number",
+            "units": u.millimeter
+        }
+    },
+    "about": (
+        "A GDAL-supported raster file containing non-zero, average "
+        "annual precipitation values for each cell."),
+    "name": "Precipitation"
+}
+ETO_ARG = {
+    "name": "Evapotranspiration Raster",
+    "type": "raster",
+    "bands": {
+        1: {
+            "type": "number",
+            "units": u.millimeter
+        }
+    },
+    "about": "A map of evapotranspiration values"
+}
+SOIL_GROUP_ARG = {
+    "type": "raster",
+    "bands": {1: {"type": "code"}},
+    "about": (
+        "Raster with values equal to 1, 2, 3, 4, corresponding to "
+        "soil hydrologic group A, B, C, or D, respectively"),
+    "name": "soil hydrologic group"
+}
+THRESHOLD_FLOW_ACCUMULATION_ARG = {
+    **GTE_0,
+    "type": "number",
+    "units": u.pixel,
+    "about": (
+        "The number of upstream cells that must flow into a cell "
+        "before it's classified as a stream."),
+    "name": "Threshold Flow Accumulation Limit"
+}
+
+LINES = {'LINESTRING', 'MULTILINESTRING'}
+POLYGONS = {'POLYGON', 'MULTIPOLYGON'}
+ALL_GEOMS = {'LINESTRING', 'POLYGON', 'POINT'}
```

Comment:
The intent here was to document which GDAL geometry types are valid for the vector input. `geometries` isn't currently used anywhere, but I'm expecting it would be useful for generating portions of the UG and expanding validation, if/when we do those things. 

I hadn't thought this through very well... These are sets of values that would be returned by `geometry.GetGeometryName()`. Maybe it should be the more specific `OGRwkbGeometryType` enum values like `ogr. wkbMultiSurfaceZ`. 

I'm assuming we only ever look at the first layer of a vector. Verifying which geometries each model accepts will be kinda tedious.